samples/emoji-search/android-views/src/main/kotlin/com/example/redwood/emojisearch/android/views/EmojiSearchActivity.kt

```diff
@@ -78,11 +81,22 @@ class EmojiSearchActivity : ComponentActivity() {
     }
 
     treehouseLayout = TreehouseLayout(this, widgetSystem, onBackPressedDispatcher).apply {
-      treehouseContentSource.bindWhenReady(this, treehouseApp)
+      treehouseContentSource.bindWhenReady(this, treehouseApp, codeListener)
     }
     setContentView(treehouseLayout)
   }
 
+  private val codeListener: CodeListener = object : CodeListener() {
+    override fun onUncaughtException(view: TreehouseView<*>, e: Throwable) {
+      val treehouseLayout = view as TreehouseLayout
```

Comment:
```suggestion
```
`treehouseLayout` is available on the instance. I would strongly prefer code listener implementations close over their precise `TreehouseView` subtype rather than doing unsafe casts. In real code this means that if you change UI toolkits (from views to Compose UI), the code below fails to compile, whereas with the unsafe cast it compiles but crashes at runtime.